internal/lint/publish.go

```diff
@@ -44,17 +45,46 @@ func Publisher(input *PublisherInput) {
 	})
 
 	ginkgo.It("should publish", func() {
-		Ω.Expect(subject.Topic(topicA).Publish(ctx, &bps.PubMessage{Data: []byte("v1")})).To(Ω.Succeed())
-		Ω.Expect(subject.Topic(topicB).Publish(ctx, &bps.PubMessage{Data: []byte("v2")})).To(Ω.Succeed())
-		Ω.Expect(subject.Topic(topicA).Publish(ctx, &bps.PubMessage{Data: []byte("v3")})).To(Ω.Succeed())
+		// run publishing and consuming concurrently (mostly for StartPosition=PositionNewest case):
+		var threads sync.WaitGroup
 
-		Ω.Eventually(func() ([]*bps.PubMessage, error) {
-			return input.Messages(topicA, 2)
-		}, "3s").Should(haveData("v1", "v3"))
+		// produce concurrently:
+		threads.Add(1)
+		go func() {
+			defer threads.Done()
+			defer ginkgo.GinkgoRecover()
 
-		Ω.Eventually(func() ([]*bps.PubMessage, error) {
-			return input.Messages(topicB, 1)
-		}, "3s").Should(haveData("v2"))
+			// give subscribers some time to actually start consuming (subscribe):
+			time.Sleep(subscriptionWaitDelay)
```

Comment:
What if we don't do this? Is there a way for us to pass a custom `func() error` ping function to the shared test for each of the providers? E.g. for Kafka we could just refresh metadata and return nil once it succeeds. just an idea. not urgent

Reply:
Basically, that should be smth like `<-allTopicsSubscribed`. I think Sleep is less correct, but much simpler for now (maybe for later as well).